### PR TITLE
fix file copy check not happening for agentless SSH connections

### LINKED ISSUE #24949: scp to agentless nodes allowed in spite of RBAC denial
Expected behavior:

When `ssh_file_copy` is set to `false` in a user's role, they are blocked from using `tsh scp` for any target node, including agentless nodes.

Current behavior:

`tsh scp` and OpenSSH `scp` both work to copy files to/from agentless nodes without error.
The action is correctly denied for teleport nodes.

I'm not sure if it's technically even possible to block file copies, but the test plan indicates the expected behaviour is to block it. So we should either fix this or update the test plan (and docs).

Bug details:
- Teleport version: v13.0.0-alpha.1
- Recreation steps: Create an agentless node. Create a role with `ssh_file_copy: false`. Assign the role to a user and try `tsh scp file user@agentless-node:`
- Debug logs

### PR DESCRIPTION
**Pull request by capnspacehook**
Fixes https://github.com/gravitational/teleport/issues/24949.

### CODE REVIEW

File: lib/srv/exec.go [hunk 1]
```diff
@@ -258,11 +249,29 @@ func (e *localExec) transformSecureCopy() error {
...
 		e.Ctx.ServerConn.RemoteAddr().String(),
 		e.Ctx.ServerConn.LocalAddr().String(),
-		strings.Join(args[1:], " "))
+		scpArgs,
+	)
 
 	return nil
 }
 
+func checkSCPAllowed(scx *ServerContext, command string) error {
```
Comment (zmb3) [inline lib/srv/exec.go:258]: Should we rename this given that we don't actually use SCP anymore?

**Review by zmb3 — COMMENTED** [review]
Is it possible to test this?

**Comment by zmb3** [comment]
Why is this going directly to branch/v13? Where is the master PR?

File: lib/srv/exec.go (hunk 1 above)
Comment (capnspacehook) [reply lib/srv/exec.go:258]: We still support it server-side unfortunately :/

**Comment by capnspacehook** [comment]
Re: merging into v13 directly, my bad it's been a long week lol I'll fix that.

Re: testing, I thought about that for awhile, not sure what to do here tbh. There are already tests asserting that `lib/srv.ServerContext.CheckSFTPAllowed` and friends work properly, but not sure how to test the rest without adding another integration test.

**Comment by capnspacehook** [comment]
Going to close and reopen to hopefully fix CI checks not starting

File: lib/srv/exec.go [hunk 2]
```diff
@@ -258,11 +249,29 @@ func (e *localExec) transformSecureCopy() error {
...
+	args := strings.Split(command, " ")
+	if len(args) == 0 {
+		return nil
+	}
+
+	// see the user is not requesting scp, return
+	_, f := filepath.Split(args[0])
+	if f != teleport.SCP {
+		return nil
+	}
```
Comment (probakowski) [inline lib/srv/exec.go:270]:
```suggestion
	if _, f := filepath.Split(args[0]); f != teleport.SCP {
		return nil
	}
```

**Review by probakowski — APPROVED** [review]

File: lib/srv/exec.go (hunk 2 above)
Comment (capnspacehook) [reply lib/srv/exec.go]: Done

**Review by lxea — APPROVED** [review]

**Pull request merged**